Bio/PDB/PDBList.py

```diff
@@ -112,6 +139,27 @@ def __init__(
         # variable for command-line option
         self.flat_tree = False
 
+    def _get_pdb_server_url(self, server: str | PDBServer | None) -> str:
+        """Get remote PDB server url.
+
+        Handle legacy server declaration (str).
+        If no server specified, chose the fastest one.
+        """
+        pdb_server = None
+        if server is None:
+            pdb_server = get_fastest_server()
+        elif isinstance(server, str):
+            return urljoin(server, "pub/pdb/")
+        elif isinstance(server, PDBServer):
+            pdb_server = server
+
+        if pdb_server:
+            return urlunsplit(
+                ("ftp", pdb_server.domain, pdb_server.pdb_directory, None, None)
```

Comment:
Again, I'd be careful about using `ftp` protocols here, since private servers might _not_ use that protocol. I'd default to https unless the `PDBServer` class explicitly points to an FTP. You can add that as a `scheme` (to match the urlsplit/unsplit nomenclature) attribute to the dataclass.

Reply:
That's not entirely accurate. If someone creates their own `PDBServer` dataclass, they'll hit this line.

I'd really go for the scheme being part of the class and, as I mentioned in my general comment, slightly refactor so that we always create a PDBServer instance at this point. It's more self-consistent.

Reply:
If of course, when someone edit the source code the software is not working as expected.

I think I see where you want to go at the end, but I think we already brought a lot of improvement in this PR, and I strongly think we must iterate step by step. 

Regarding the current implementation, we did not

Reply:
Biopython is often used as a library, so someone could look at your nice `PDBServer` dataclass and say "awesome, I'll write one for my own custom server" and then they run into trouble. I agree that there are a few changes here already, but since you're introducing the `PDBServer` class here, we can make it more extensible from the get-go so that when we move on to a more in-depth refactoring/rethinking of this module, we have less work :)

I trust you! I just want to make sure we're on the same page here before committing changes. If you truly feel this is out of scope for this PR, then I accept your argument and we can move on.

Reply:
I was feeling like the conversation was stuck, obviously, I was wrong. Thank you for your effort.

Yes, the objective is to move from `PDBList.pdb_server: str` to `PDBList.pdb_server: PDBServer` in order to enjoy the future method of the dataclass. But I think it deserves a dedicated PR, I took interested in this work (as you noticed) and I would like to continue in an iterative way.

Reply:
Alright, let's iterate then!